lib/omniauth/strategies/openid_connect.rb

```diff
@@ -231,7 +233,7 @@ def access_token
       end
 
       def decode_id_token(id_token)
-        ::OpenIDConnect::ResponseObject::IdToken.decode(id_token, public_key)
+        ::OpenIDConnect::ResponseObject::IdToken.decode(id_token, public_key_or_config)
```

Comment:
@nov In https://github.com/omniauth/omniauth_openid_connect/pull/134, I'm fixing how this gem handles  HS256 signatures. I'm wondering if you see any issues with the approach there. 

I was actually just looking at the changes made in the `openid-connect` gem in https://github.com/omniauth/omniauth_openid_connect/pull/134/files#r1027569274. I think that change assumes the JWT includes the `kid` when HS256 is used. It's been a while since I've played with Keycloak, but if I recall correctly it may not be included. I'll have to try this again.